scripts/views/mapView.js

```diff
@@ -1,439 +1,391 @@
+      } else {
+        var visibility = map.getLayoutProperty('selected-fill', 'visibility');
+        if (visibility === 'visible') {
+          map.setLayoutProperty('selected-fill', 'visibility', 'none');
+          map.setLayoutProperty('selected-border', 'visibility', 'none');
+        }
+      }
+    });
+  }
+
+  // Fetch data from Firebase, run map filter & point layers
+  function readData () {
+    var townHallsFB = firebase.database().ref('/townHalls/').once('value').then(function(snapshot){
+      var ele = new TownHall (snapshot.val());
+      townhallData = ele
+
+      filterMap(ele);
+      makePointLayer(ele);
+      makeTable(ele);
+    });
   };
 
-  window.onResizeMap = function onResizeMap() {
-    var geocoder = new google.maps.Geocoder();
-    var resizeBounds = new google.maps.LatLngBounds();
-    var data = TownHall.isCurrentContext ? TownHall.currentContext:TownHall.allTownHalls;
-    if ( TownHall.zipQuery) {
-      resizeBounds.extend(TownHall.zipQuery);
-    }
-    data.forEach(function(ele){
-      if (ele.lat && ele.lng) {
-        marker = new google.maps.LatLng(ele.lat, ele.lng);
-        resizeBounds.extend(marker);
+  // Creates the point layer.
+  function makePointLayer (data) {
+    var featuresHome = {
+      type: 'FeatureCollection',
+      features: []
+    };
+
+    for (key in data){
+      var type = data[key].meetingType;
+
+      if (type === 'Teletown Hall' || type === 'Tele-town Hall'){
+        var iconKey = 'phone-in';
       } else {
-        // console.log(ele.eventId);
+        var iconKey = 'in-person';
+      }
+
+      if (data[key].lat) {
+        featuresHome.features.push({
+          type: 'Feature',
+          geometry: {
+            type: 'Point',
+            coordinates: [data[key].lng, data[key].lat]
+          },
+          properties: {
+            district: data[key].District,
+            icon: iconKey
+          },
+        });
+      }
+    }
+
+    map.addLayer({
+      'id': 'townhall-points',
+      'type': 'symbol',
+      'source': {
+        'type': 'geojson',
+        'data': featuresHome
+      },
+      'layout': {
+        'icon-image': '{icon}',
+        'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
+        'icon-ignore-placement': true,
+        'icon-ignore-placement': true,
+        'icon-offset': {
+          'base': 1,
+          'stops': [
+              [0, [0, -15]],
+              [10, [0, -10]],
+              [12, [0, 0]]
+          ]
+        }
       }
     });
-    google.maps.event.trigger(map, 'resize');
-    map.fitBounds(resizeBounds);
-  };
+  }
+
+  // Adds a Popup listener to the point layer. TODO: Determine content for the popup.
+  function addPopups () {
+    var popup = new mapboxgl.Popup({
+      closeButton: false,
+      closeOnClick: false
+    });
+
+    map.on('mousemove', function(e) {
+      var features = map.queryRenderedFeatures(e.point, { layers: ['townhall-points'] });
```

Comment:
When the event cards ("sidebar") aren't present the popups should contain the Rep, the place, and the date.

When the event cards ("sidebar") is open let's nix the popups all together and just highlight the corresponding event card (scroll it into view if needed).  No point in having redundant data.  Bonus points if you make mousing over the MoC card highlight the event on the map.